File: src/main/java/com/rpb/reservation/reservation/application/ReservationPublicShare.java
package com.rpb.reservation.reservation.application;

public record ReservationPublicShare(
    String reservationNo,
    String storeName,
    String reservationDate,
    String reservationTime,
    int partySize,
    String tableCode,
    boolean tablePending,
    String arrivalNote,
    String storePhone,
    String storeEmail,
    String storeWhatsappPhone,
    String storeAddress,
    String googleMapUrl,
    String shareTitle,
    String shareSummary,
    String shareText
) {
}

File: src/types/reservationPublicShare.ts
export interface ReservationPublicShare {
  reservationNo: string
  storeName: string
  reservationDate: string
  reservationTime: string
  partySize: number
  tableCode: string
  tablePending: boolean
  arrivalNote: string
  storePhone: string
  storeAddress: string
  googleMapUrl: string
  shareTitle: string
  shareSummary: string
  shareText: string
}

export interface ReservationPublicShareResponse {
  success: true
  share: ReservationPublicShare
}

export interface ReservationPublicShareApiErrorResponse {
  success: false
  error: {
    code: string
    messageKey: string
    details: Record<string, unknown>
  }
}
